pkg/provisioner/ironic/ironic.go

```diff
@@ -931,6 +931,26 @@ func (p *ironicProvisioner) Deprovision(deleteIt bool) (result provisioner.Resul
 			nodes.ProvisionStateOpts{Target: nodes.TargetManage},
 		)
 
+	case nodes.Inpsecting:
+		p.log.Info("waiting for inspection to complete")
+		result.Dirty = true
+		result.RequeueAfter = introspectionRequeueDelay
+		return result, nil
+
+	case nodes.InspectWait:
```

Comment:
FYI if the node is in InspectWait, it still can successfully finish inspection and go to manageable

Reply:
When we're deleting the host in k8s, we want to tell Ironic to stop doing whatever it is doing, power the node off, and then delete the node record. We don't want it to finish, because nothing is going to use the data.